### DISCUSSION TITLE
insert my data to paitition

### DISCUSSION THREAD

**Discussion by AriesJin** [milvus-io/milvus#37883] [Q&A and General discussion] [answered]
I want to insert my data to partA, but all data only insert in _default, why?
here is my code.

# init
from pymilvus import MilvusClient, DataType
import time
client = MilvusClient(
uri="http://*****"

# Create schema
schema = MilvusClient.create_schema(
    auto_id=False,
    enable_dynamic_field=True,
)

# Add fields to schema
schema.add_field(field_name="my_id", datatype=DataType.INT64, is_primary=True)
schema.add_field(field_name="my_vector", datatype=DataType.FLOAT_VECTOR, dim=6)

index_params = client.prepare_index_params()
index_params.add_index(
    field_name="my_id",
    index_type="STL_SORT"
)
index_params.add_index(
    field_name="my_vector", 
    index_type="IVF_FLAT",
    metric_type="IP",
    params={ "nlist": 128 }
)

client.create_collection(
    collection_name="test_col",
    schema=schema,
    index_params=index_params
)

# insert data & partition_names="partA",
data = [
    {"my_id": 1, "my_vector": [1.3333, -22.2, 0.2, -0.2, 90.22, 3.9], "color": "pink_8682"},
    {"my_id": 2, "my_vector": [3.2000, 0.2, 0.2, 0.22, 0.33, 9.33], "color": "red_7025"}
]
res = client.insert(
collection_name="test_ez",
partition_names="partA",
data=data
)

# query in partition_names="_default"
res = client.query(
    collection_name="test_col",
    filter="",  #
    output_fields=["*"],  # 
    partition_names=[ "_default"],  # 
    limit=1000  # 
)
## result of "_default"
print(res)
"""
output 
data: ["{'my_id': 1, 'my_vector': [1.3333, -22.2, 0.2, -0.2, 90.22, 3.9], 'color': 'pink_8682'}", "{'my_id': 2, 'my_vector': [3.342, 0.2, 0.2, 0.22, 0.33, 9.33], 'color': 'red_7025'}"] 
"""
# query in partition_names="partA"
res = client.query(
    collection_name="test_col",
    filter="",  # 
    output_fields=["*"],  # 
    partition_names=[ "partA"],  # 
    limit=1000  # 
)
## result of "partA"
print(res)
"""
output 
data: [] 
"""

**Comment by yhmo** [comment, marked as answer]
Each call of insert() only can target one partition. So, the parameter is "partition_name", not "partition_names".
```
res = client.insert(
  collection_name="xxx",
  partition_name="partA",
  data=data
)
```

Each call of search()/query() can target multiple partitions, so the parameter is "partition_names", not "partition_name".

```
res = client.query(
  collection_name="xxx",
  filter="", 
  output_fields=["*"], 
  partition_names=[ "_default", "partA", "partB"],
)
```

**Reply by AriesJin** [reply]
thanks! it's work  for me!

i have another question!
if partitionKey is used, does this mean that all the data is loaded? If so, then partitionKey is kind of the same as trading memory for retrieval speed.

If I use partition then I need to manage which partition name to load at this time for different users. It is also necessary to release it after not using it for a long time. Is there a better solution?

anyone help me?
